pwndbg/commands/hijack_fd.py

```diff
@@ -0,0 +1,283 @@
+from __future__ import annotations
+
+import argparse
+import contextlib
+import socket
+from typing import Literal
+from typing import NamedTuple
+from typing import Optional
+from typing import Tuple
+from typing import TypedDict
+from urllib.parse import ParseResult
+from urllib.parse import urlparse
+
+import pwndbg.aglib.memory
+import pwndbg.aglib.shellcode
+import pwndbg.commands
+import pwndbg.lib.abi
+import pwndbg.lib.memory
+import pwndbg.lib.regs
+from pwndbg.commands import CommandCategory
+
+
+class ShellcodeRegs(TypedDict):
+    newfd: str
+    syscall_ret: str
+    stack: str
```

Comment:
Why not just a dataclass? They don't need a dict and are likely faster?

Reply:
maybe NamedTuple then :)